### ISSUE TITLE
SublimeLinter-eslint not working when eslint es installed locally

### ISSUE THREAD

**Issue by doberkofler** [SublimeLinter/SublimeLinter3#191]
As suggested by @roadhump in issue 23 of SublimeLinter-eslint (https://github.com/roadhump/SublimeLinter-eslint/issues/23) I open this issue here.

I just noticed, that when eslint is also installed locally the following error is reported:
```
SublimeLinter: eslint: ods.ui.form.js ['D:\\MyDev\\ljs_app\\trunk\\periscope\\build\\node_modules\\.bin\\eslint', '--format=compact', '--stdin'] 
SublimeLinter: ERROR: could not launch ['D:\\MyDev\\ljs_app\\trunk\\periscope\\build\\node_modules\\.bin\\eslint', '--format=compact', '--stdin'] 
SublimeLinter: reason: [WinError 193] %1 is not a valid Win32 application 
```
I'm running ST3 Build 3065 on a Windows 7 x64 system and when eslint is only installed globally, the linter plugin works as expected. I'm using eslint 0.14.0.

I also noticed that (other than the jscs and jshint) plugins eslint is not shown when loading ST.
I just wanted to mention it but it actually should not be related to this problem as it is also not shown when eslint is only installed globally and the plugin is working:
```
SublimeLinter: jscs version query: C:\Users\doberkofler.LBITS\AppData\Roaming\npm\jscs.cmd --version 
SublimeLinter: jscs version: 1.10.0 
SublimeLinter: jscs: (>= 1.0.10) satisfied by 1.10.0 
SublimeLinter: jscs activated: C:\Users\doberkofler.LBITS\AppData\Roaming\npm\jscs.cmd 
SublimeLinter: jshint version query: C:\Users\doberkofler.LBITS\AppData\Roaming\npm\jshint.cmd --version 
SublimeLinter: jshint version: 2.6.0 
SublimeLinter: jshint: (>= 2.5.0) satisfied by 2.6.0 
SublimeLinter: jshint activated: C:\Users\doberkofler.LBITS\AppData\Roaming\npm\jshint.cmd 
SublimeLinter: found existing HTML syntax, version 3 
SublimeLinter: found existing Rails syntax, version 1 
```

**Comment by skj3gg** [comment]
I'm having a hard time replicating this. So you have eslint installed locally but not globally?

**Comment by doberkofler** [comment]
I actually have it installed globally and locally. Until now I did not even know that SublimeLinter3
 is able to access a local installation.
The error happens when there is a local installation. If I remove the local installation everything works as expected.

**Comment by skj3gg** [comment]
I figured out what the problem is, there is no .cmd at the end of the command. When python searches `D:\\MyDev\\ljs_app\\trunk\\periscope\\build\\node_modules\\.bin` it finds the first file that satisfies `eslint` but that isn't actually a windows executable. the actual file it wanted was the second one, `eslint.cmd` 

I'll try to resolve this but don't really use node or eslint so give me a bit, unless you would like to contribute of course!

**Comment by doberkofler** [comment]
I'm happy to help, if I can be of any assistance!
Looking at the python code, I would guess that the problem is in the function `find_local_cmd_path` in `lint/node_linter.py`. Am I right ?

**Comment by doberkofler** [comment]
I'm sure there is better ways to code this, but generally speaking, the following 3 lines of code in the function `find_local_cmd_path` in the file `lint/node_linter.py` after the line `binary = path.join(node_modules_bin, cmd)` would solve the problem:

```python
        import sys
        if sys.platform.startswith('win'):
            binary += '.cmd'
```

**Comment by skj3gg** [comment]
On the road, so may not work but there is `sublime.platform()` should return `'windows'`

**Comment by johnmcase** [comment]
+1

**Comment by doberkofler** [comment]
it's ugly as hell and you should definitely use `sublime.platform()` but the code I posted does actually solve the problem, if you need a quick fix.

**Comment by skj3gg** [comment]
This has been released and should be visible in 30 minutes.

Thanks!

**Comment by remitbri** [comment]
I actually have the same problem -eslint installed both locally and globally-, but I'm on OSX (10.9 and 10.10). ST3 build 3073. The linter works again if I move `./node_modules/.bin/eslint` 
````console
SublimeLinter: json activated: <builtin> 
reloading Packages/User/SublimeLinter.sublime-settings
SublimeLinter: eslint: index.jsx ['/Users/myUserName/path/to/myProject/node_modules/.bin/eslint', '--format=compact', '--stdin'] 
SublimeLinter: eslint output:
module.js:338
    throw err;
          ^
Error: Cannot find module 'concat-stream'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/Users/myUserName/path/to/myProject/node_modules/.bin/eslint:2:14)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10) 
````

**Comment by skj3gg** [comment]
Actually this doesn't look like the same problem. Yours is working and loaded into sublime. The problem seems to exist in your version of Eslint. The error in the console is a JavaScript error, for which SL doesn't have any JavaScript code. So that error is being produced by Eslint.
